hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java

```diff
@@ -151,208 +149,238 @@ public PreHandleContext status(@NonNull final ResponseCodeEnum status) {
+            // keys? Or KeyList with Contract keys only?
+            throw new PreCheckException(responseCode);
+        }
+
+        return requireKey(key);
     }
 
     /**
-     * Checks if the contractId is a sentinel id <code>0.0.0</code>
+     * Creates a new {@link PreHandleContext} for a nested transaction. The nested transaction will be set on
+     * this context as the "inner context". There can only be one such at a time. The inner context is returned
+     * for convenience.
      *
-     * @param id given contractId
-     * @return true if the given contractId is
+     * @param nestedTxn the nested transaction
+     * @param payerForNested the payer for the nested transaction
+     * @param responseCode the response code to be used if a {@link PreCheckException} is thrown
+     * @return the inner context
+     * @throws PreCheckException If the payer is not valid
      */
-    private boolean designatesContractRemoval(@NonNull final ContractID id) {
-        return id.shardNum() == 0 && id.realmNum() == 0 && id.contractNumOrElse(0L) == 0 && !id.hasEvmAddress();
+    @NonNull
+    public PreHandleContext createNestedContext(
+            @NonNull final TransactionBody nestedTxn,
+            @NonNull final AccountID payerForNested,
+            @NonNull final ResponseCodeEnum responseCode)
+            throws PreCheckException {
+        this.innerContext = new PreHandleContext(accountAccess, nestedTxn, payerForNested, responseCode);
+        return this.innerContext;
     }
 
     /**
-     * Given a successful key lookup, adds its key to the required signers. Given a failed key lookup, sets this
-     * {@link PreHandleContext}'s status to either the failure reason of the lookup; or (if it is non-null), the
-     * requested failureStatus parameter.
+     * Gets the inner context, if any.
      *
-     * @param result key lookup result
-     * @param failureStatus failure reason for the lookup
+     * @return The inner context.
      */
-    private void addToKeysOrFail(
-            final KeyOrLookupFailureReason result,
-            @Nullable final ResponseCodeEnum failureStatus,
-            final boolean isPayer) {
-        if (result.failed()) {
-            this.status = failureStatus != null ? failureStatus : result.failureReason();
-        } else if (result.key() != null) {
-            if (isPayer) {
-                this.payerKey = result.key();
-            } else {
-                this.requiredNonPayerKeys.add(result.key());
-            }
-        }
+    public PreHandleContext innerContext() {
+        return innerContext;
     }
 
-    @NonNull
-    public PreHandleContext createNestedContext(
-            @NonNull final TransactionBody nestedTxn, @NonNull final AccountID payerForNested) {
-        return new PreHandleContext(accountAccess, nestedTxn, payerForNested);
+    @Override
+    public String toString() {
+        return "PreHandleContext{" + "accountAccess="
```

Comment:
nitpick, but this would likely read a lot better if it was a text block and `.formatted()`
e.g.
```java
return """
    PreHandleContext{accountAccess="%s", txn="%s", payer="%s", requiredNonPayerKeys="%s", \
    status="%s", innerContext="%s"}\
    """.formatted(accountAccess, txn, payer, requiredNonPayerKeys, payerKey, innerContext);
```

Reply:
Personally I'm fine with the way it is, but in any event, this code was here before (diff just thinks it is new). I'm not sure why we even have it, TBH. I'm not a big fan of overriding toString in general, probably because I prefer debuggers over toString debuggin